Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DocumentPropertyEditor.cpp

```diff
@@ -0,0 +1,310 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#include "DocumentPropertyEditor.h"
+
+#include <QLabel>
+#include <QLineEdit>
+#include <QVBoxLayout>
+
+#include <AzCore/DOM/DomUtils.h>
+#include <AzFramework/DocumentPropertyEditor/PropertyEditorNodes.h>
+
+namespace AzToolsFramework
+{
+    // helper method used by both the view and row widgets to empty layouts
+    static void destroyLayoutContents(QLayout* layout)
```

Comment:
Uber nit: I think our style preference is upper-camel case even though all the Qt APIs are camel case.